test/benchmarks/benchmark_png_encode.cc

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include <benchmark/benchmark.h>
+#include "png_funcs.h"
+#include <assert.h>
+
+#define IMWIDTH 1024
+#define IMHEIGHT 1024
+
+class png_encode: public benchmark::Fixture {
+private:
+    png_dat outpng;
+
+    /* Backing this on the heap is a more realistic benchmark */
+    uint8_t *input_img_buf = NULL;
+
+public:
+    /* Let's make the vanilla version have something extremely
+     * compressible */
+    virtual void init_img(png_bytep img_bytes, size_t width, size_t height) {
+        init_compressible(img_bytes, width*height);
+    }
+
+    void SetUp(const ::benchmark::State& state) {
+        input_img_buf = (uint8_t*)malloc(IMWIDTH * IMHEIGHT * 3);
+        outpng.buf = (uint8_t*)malloc(IMWIDTH*IMHEIGHT*3);
```

Comment:
Inconsistent malloc size formatting. Use spaces between operators on second line like on the first. Also on the `outpng.buf_rem` assignment.